lib/test.js

```diff
@@ -0,0 +1,261 @@
+    }
+
+    // no error but one expected
+    return {
+      passed: false,
+      message: getFailErrorMessage({
+        ruleName: rule.ruleName,
+        functionName: rule.rule.functionName,
+        functionsFile: rule.functionsFile,
+        exampleTitle: example.meta.title,
+        exampleFile: example.file,
+        expected: example.meta.error,
+        actual: '',
+        target: target || example.meta.target,
+        type,
+      }),
+    };
+  }
+
+  // no error expected but got one
+  if ((_.isArray(VkResult) && VkResult.length !== 0) || vkError) {
+    return {
+      passed: false,
+      message: getFailErrorMessage({
+        ruleName: rule.ruleName,
+        functionName: rule.rule.functionName,
+        functionsFile: rule.functionsFile,
+        exampleTitle: example.meta.title,
+        exampleFile: example.file,
+        expected: '',
+        actual,
+        target: target || example.meta.target,
+        type,
+      }),
+    };
+  }
+
+  // no error expected , got no error
+  return {
+    passed: true,
+    message: getPassedErrorMessage({
+      ruleName: rule.ruleName,
+      exampleTitle: example.meta.title,
+      type,
+      target: target || example.meta.target,
+    }),
+  };
+}
+
+let atLeastOneFailed = false;
+const failed = [];
+
+
+helpers.getApiaryToken((err, apiaryToken) => {
+  if (err) {
+    console.error(err.message || err);
+    process.exit(1);
+  }
+
+  function getTargets(example) {
+    let targets = _.get(example, 'meta.target', []);
+
+    if (!_.isArray(targets)) {
+      targets = [targets];
+    }
+
+    return targets;
+  }
+
+  const tests = [];
+
+  _.each(_.get(styleGuides, 'rules', []), (rule) => {
+    _.each(_.get(rule, 'examples.good', []), (example) => {
+      _.each(getTargets(example), (target) => {
+        tests.push({
+          rule,
+          example,
+          type: 'good',
+          target,
+        });
+      });
+    });
+    _.each(_.get(rule, 'examples.bad', []), (example) => {
+      _.each(getTargets(example), (target) => {
+        tests.push({
+          rule,
+          example,
+          type: 'bad',
+          target,
+        });
+      });
+    });
+  });
+
+
+  let currentRule = '';
+
+  async.eachSeries(tests, (test, next) => {
+    helpers.testExample({
+      functions,
+      example: test.example,
+      functionName: _.get(test, 'rule.rule.functionName'),
+      minim: _.get(test, 'rule.rule.minim'),
+      target: test.target,
+      apiaryToken,
+    }, (error, results, target) => {
+      if (error) {
+        return next(error);
+      }
+      const result = getResult({
+        VkResult: results,
+        rule: test.rule,
+        example: test.example,
+        type: test.type,
+        target,
+      });
+
+      if (!result.passed) {
+        atLeastOneFailed = true;
+        failed.push(result.message);
+      }
+
+      if (currentRule !== test.rule.ruleName) {
+        currentRule = test.rule.ruleName;
+        console.log('');
```

Comment:
This just jumps an empty line - we might want to exchange it for a progress indicator? (_`x of y`_ ?)

Reply:
it is this dividing line 
![snap513](https://user-images.githubusercontent.com/752248/38369665-8dcdd15e-38e8-11e8-8d64-99d93d671304.png)